### PR TITLE
Lidar hotfix 1 april 2018

### PR DESCRIPTION
**Pull request by stevempope**
Applies the fixes we identified at the last Hardware Day

Fixes increment issues and pointer issues.

### CODE REVIEW

**Comment by stevempope** [comment]
Added LSenseTest method

**Review by RyanShoobert — APPROVED** [review]

File: mcomp_C/lsensor/LSensor.cpp [hunk 1]
```diff
@@ -110,14 +110,14 @@ unsigned int* LSensor::decodeRead(){
 
 bool LSensor::adjustRPM(){
   getAvgRPM();
-  if (avgRPM > targetRPM+10){ //if RPM is more than 10RPM off target, adjust it
-    targetPWM = targetPWM-5;
+  if (avgRPM > targetRPM+20){ //if RPM is more than 10RPM off target, adjust it
+    targetPWM = targetPWM--;
```
Comment (SamanthaDawsonBanks) [inline mcomp_C/lsensor/LSensor.cpp]: should be
a=a-1
or 
a--

**Review by SamanthaDawsonBanks — APPROVED** [review]

**Comment by SamanthaDawsonBanks** [comment]
approved pending typo fix

File: mcomp_C/lsensor/LSensor.cpp (hunk 1 above)
Comment (stevempope) [reply mcomp_C/lsensor/LSensor.cpp]: This has been fixed now, merging

**Pull request merged**